openpype/hosts/houdini/plugins/publish/validate_primitive_hierarchy_paths.py

```diff
@@ -103,4 +113,62 @@ def get_invalid(cls, instance):
                 "Prims have no value for attribute `%s` "
                 "(%s of %s prims)" % (path_attr, len(invalid_prims), num_prims)
             )
-            return [output_node.path()]
+            return [output_node]
+
+    @classmethod
+    def repair(cls, instance):
+        """Add a default path attribute Action.
+
+        it's a helper action more than a repair action,
+        which used to add a default single value.
+        """
+
+        rop_node = hou.node(instance.data["instance_node"])
+        output_node = rop_node.parm('sop_path').evalAsNode()
+
+        if not output_node:
+            cls.log.debug(
+                "Action isn't performed, empty SOP Path on %s"
+                % rop_node
+            )
+            return
+
+        # This check to prevent the action from running multiple times.
+        # git_invalid only returns [output_node] when
+        #   path attribute is the problem
+        if cls.get_invalid(instance) != [output_node]:
+            return
+
+        path_attr = rop_node.parm("path_attrib").eval()
+
+        path_node = output_node.parent().createNode("name", "AUTO_PATH")
+        path_node.parm("attribname").set(path_attr)
+        path_node.parm("name1").set('`opname("..")`/`opname("..")`Shape')
+
+        cls.log.debug(
+            "'%s' was created. It adds '%s' with a default single value"
+            % (path_node, path_attr)
+        )
+
+        path_node.setGenericFlag(hou.nodeFlag.DisplayComment,True)
+        path_node.setComment(
+            'Auto path node created automatically by "Add a default path attribute"'
+            '\nFeel free to modify or replace it.'
+        )
+
+        if output_node.type().name() in ['null', 'output']:
+            # Connect before
+            path_node.setFirstInput(output_node.input(0))
+            path_node.moveToGoodPosition()
+            output_node.setFirstInput(path_node)
+            output_node.moveToGoodPosition()
+        else:
+            # Connect after
+            path_node.setFirstInput(output_node)
+            rop_node.parm('sop_path').set(path_node.path())
+            path_node.moveToGoodPosition()
+
+            cls.log.debug(
+                "SOP path on '%s' updated to new output node '%s'"
+                % (rop_node, path_node)
+            )
```

Comment:
Since you're already using `%s` formatting I think you can actually allow the logger to format it only if a handler uses it by changing this to:
```suggestion
            cls.log.debug(
                "SOP path on '%s' updated to new output node '%s'",
                rop_node, path_node
            )
```
This would only really be faster if there's no logging handler using it however - but it might be more readable and due to it potentially being faster maybe a better standard too?

@iLLiCiTiT what do you think?

Reply:
Interesting, I didn't know that!